### DISCUSSION TITLE
【请教】 UI设计的.pag文件，不能正常的显示(PC端)

### DISCUSSION THREAD

**Discussion by hubeishuaige** [Tencent/libpag#1985] [Q&A]
我们的UI按照官网的教程，导出来的.pag文件，web(pc)端用JS  PAG.PAGComposition.make() 创建了一个空的PAGComposition，然后加载设计师设计的.pag文件，这个时候控制台也没有报错，但是页面是上面都没有，一片空白。附上我的代码：

`PAGInit().then(async (PAG) => {
                const canvasEle = document.createElement('canvas');
                const renderCanvas = new OffscreenCanvas(405, 720);
                canvasEle.id = 'canvasPag'
                canvasEle.width = 405;
                canvasEle.height = 720;
                const ctx = canvasEle.getContext('2d');
                document.getElementById('monitorInner').appendChild(canvasEle);

                const textUrl = './text.pag'; // 渲染文字 -- 这个是demo里面的素材
                const file = await fetch(textUrl).then((response) => response.arrayBuffer());
                const textFile = await PAG.PAGFile.load(file);

                const imageUrl = './testB.pag'; // 渲染图片 --- 这个是我们UI设计的素材
                const file2 = await fetch(imageUrl).then((response) => response.arrayBuffer());
                const imageFile = await PAG.PAGFile.load(file2);

                const pagComposition = PAG.PAGComposition.make(imageFile.width(), imageFile.height());
                
                pagComposition.addLayer(textFile); // 添加文本
                pagComposition.addLayer(imageFile); // 添加图片

                const pagView = await PAG.PAGView.init(pagComposition, renderCanvas);
                // 渲染画布上的内容
                pagView?.addListener('onAnimationFlushed', () => {
                    ctx?.clearRect(0, 0, canvasEle.width, canvasEle.height);
                    ctx?.drawImage(
                        renderCanvas,
                        0,
                        0,
                        renderCanvas.width,
                        renderCanvas.height,
                        0,
                        0,
                        canvasEle.width,
                        canvasEle.height,
                    );
                });

                pagView.setRepeatCount(0); // 设置动画重复的次数。默认值是1，这意味着动画将只播放一次。0表示动画将播放无限次。
                await pagView.play();
            });`

但是如果把设计师设计的.pag文件和demo里面的.pag文件一起用 addLayer方法添加进来，则能正常的查看，如果只添加设计师设计的.pag文件，就是一片空白。。我们设计师设计的文件下载地址：https://shuziyunxie-1313466856.cos.ap-guangzhou.myqcloud.com/picture_url/3db0d3d5-bded-4e1e-8a72-6ae744e40973.pag  谁可以帮忙看看，非常感谢

示例代码包：执行  yarn dev 即可运行  或者  npm  run dev
[pag-demo.zip](https://github.com/Tencent/libpag/files/13645405/pag-demo.zip)

**Comment by domchen** [comment]
你断点检查一下这句：const pagComposition = PAG.PAGComposition.make(imageFile.width(), imageFile.height()); 传入的宽高是不是不正常。

**Reply by hubeishuaige** [reply]
宽是1701  高是  2551 一切看的都是正常的，狠是纳闷

**Reply by domchen** [reply]
弄个可以运行的demo传上来吧，代码里看不出来什么问题。

**Reply by hubeishuaige** [reply]
好的，我去做个demo上传上来。。稍等

**Reply by hubeishuaige** [reply]
大湿兄，我做了一个demo，方便的时候帮我看看，不胜感激

**Comment by CodeJhF** [comment]
这条我们在当时的回复里已经给出结论，这里就不重复贴了，先关闭本条。
如果按当时的结论排查后仍有问题，欢迎回复补充最小复现 demo 与版本号（4.4 / 4.5），我们会重新看。（stale-close）
